volatility/framework/layers/crash.py

```diff
@@ -166,11 +167,15 @@ def _load_segments(self) -> None:
                         segments.append((start_position * 0x1000, mapped_offset, length, length))
                         mapped_offset += length
 
+
                 # Finish it off
-                if bit_position == (len(summary_header.BufferLong) * 32) - 1 and current_bit == 1:
+                if (bit_position == (len(summary_header.BufferLong) * 32) - 1 or bit_position == (len(summary_header.BufferLong) * 32) - 1 -32*0x2000) and current_bit == 1:
```

Comment:
This is now a very complex expression, so I'd prefer it were cut down to make it more easily readable and again documented to explain what the changes are for.